src/test/java/org/takes/http/BkReuseTest.java

```diff
@@ -0,0 +1,170 @@
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.http;
+
+import com.google.common.base.Joiner;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.util.regex.Pattern;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.takes.tk.TkGreedy;
+import org.takes.tk.TkText;
+
+/**
+ * Test case for {@link BkReuse}.
+ *
+ * @author Piotr Pradzynski (prondzyn@gmail.com)
+ * @version $Id$
+ * @checkstyle MultipleStringLiteralsCheck (500 lines)
+ * @todo #306 Implement missing BkReuse.accept method to support
+ *  HTTP persistent connections and to pass below three tests.
+ *
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class BkReuseTest {
+
+    /**
+     * BkReuse can handle two requests in one connection.
+     * @throws Exception If some problem inside
+     */
+    @Ignore
+    @Test
+    public void handlesTwoRequestInOneConnection() throws Exception {
+        final String text = "Hello world!";
+        final Socket socket = Mockito.mock(Socket.class);
+        Mockito.when(socket.getInputStream()).thenReturn(
+            new ByteArrayInputStream(
+                Joiner.on("\r\n").join(
+                    "POST / HTTP/1.1",
+                    "Host: localhost",
+                    "Content-Length: 4",
+                    "",
+                    "hi",
+                    "POST / HTTP/1.1",
+                    "Host: localhost",
+                    "Content-Length: 4",
+                    "",
+                    "hi"
+                ).getBytes()
+            )
+        );
+        Mockito.when(socket.getLocalAddress()).thenReturn(
+            InetAddress.getLocalHost()
+        );
+        Mockito.when(socket.getLocalPort()).thenReturn(0);
+        Mockito.when(socket.getInetAddress()).thenReturn(
+            InetAddress.getLocalHost()
+        );
+        Mockito.when(socket.getPort()).thenReturn(0);
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Mockito.when(socket.getOutputStream()).thenReturn(baos);
+        new BkReuse(new BkBasic(new TkGreedy(new TkText(text)))).accept(socket);
+        final Pattern pattern = Pattern.compile(
+            text + ".*?" + text,
+            Pattern.DOTALL
+        );
+        Assert.assertTrue(
```

Comment:
@prondzyn Let's use Hamcrest matchers for consistency. You can use `RegexMatchers` from [jcabi-matchers](http://matchers.jcabi.com/apidocs-1.3/com/jcabi/matchers/RegexMatchers.html).